lib/segment/src/index/field_index/full_text_index/mutable_inverted_index.rs

```diff
@@ -135,34 +144,47 @@ impl MutableInvertedIndex {
         intersect_postings_iterator(postings)
     }
 
-    pub fn values_count(&self, point_id: PointOffsetType) -> usize {
-        // Maybe we want number of documents in the future?
-        self.get_doc(point_id).map(|x| x.len()).unwrap_or(0)
+    fn get_posting_len(&self, token_id: TokenId) -> Option<usize> {
+        self.postings
+            .get(token_id as usize)
+            // unwrap safety: token_id is a valid index. Same case as in `filter` method.
```

Comment:
After extracting this snippet of code from `estimate_cardinality` into a separate trait method, this assumption is more error prone as it assumes the only callsite calls it correctly.

I suggest either to make it return an `Err` or `None` (maybe with a `debug_assert`) instead of a panicking unwrap, or at least put a warning into the trait method doc, e.g.:

```rust
pub trait InvertedIndex {
    …

    /// # Panics
    /// Panics when `token_id` is not a valid index.
    fn get_posting_len(&self, token_id: TokenId) -> Option<usize>;

    …
}
```

Reply:
Tbh the trait was not strictly necessary since we never request a trait object or an impl Trait. But it does make implementing all dispatched methods easier. 

I agree that we should document the panic, but even better, we can just return `None` :)